webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java

```diff
@@ -382,9 +452,17 @@ private SmartlingFile processTranslationBatch(List<TextUnitDTO> batch,
 
         if (!options.isDryRun()) {
             logger.debug("Push Android file to Smartling project: {} and locale: {}", projectId, localeTag);
-            smartlingClient.uploadLocalizedFile(projectId, sourceFilename, "android",
+            Mono.fromCallable(() -> smartlingClient.uploadLocalizedFile(projectId, sourceFilename, "android",
                     getSmartlingLocale(localeMapping, localeTag), file.getFileContent(),
-                    options.getPlaceholderFormat(), options.getCustomPlaceholderFormat());
+                    options.getPlaceholderFormat(), options.getCustomPlaceholderFormat()))
+                    .retryWhen(Retry.backoff(3, Duration.ofSeconds(1))
```

Comment:
should we have a function that create the default retry since it is repeated a few times? also retrying right away the first time might be better (eg. for authentication issue ... though that should be fixed now) and have a max backoff of 1 second (`.maxBackoff(Duration.ofSeconds(1))`)?